Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2248,14 +2250,31 @@ static IntRect computeMarqueeSelectionRect(const WebCore::IntPoint& point1, cons
     return m_currentSelection.get().string;
 }
 
-bool UnifiedPDFPlugin::existingSelectionContainsPoint(const FloatPoint&) const
+bool UnifiedPDFPlugin::existingSelectionContainsPoint(const FloatPoint& rootViewPoint) const
 {
-    return false;
+    auto documentPoint = convertFromPluginToDocument(convertFromRootViewToPlugin(IntPoint { rootViewPoint }));
+
+    auto pageIndex = pageIndexForDocumentPoint(documentPoint);
+    if (!pageIndex)
+        return false;
+
+    auto page = m_documentLayout.pageAtIndex(*pageIndex);
+    return IntRect { [m_currentSelection boundsForPage:page.get()] }.contains(convertFromDocumentToPage(documentPoint, *pageIndex));
 }
 
-FloatRect UnifiedPDFPlugin::rectForSelectionInRootView(PDFSelection *) const
+FloatRect UnifiedPDFPlugin::rectForSelectionInRootView(PDFSelection *selection) const
 {
-    return { };
+    if (!selection || !selection.pages)
+        return { };
+
+    RetainPtr page = [selection.pages firstObject];
+    auto pageIndex = m_documentLayout.indexForPage(page);
+    if (!pageIndex)
+        return { };
+
+    auto rectInPageSpace = IntRect { [selection boundsForPage:page.get()] };
+    IntRect rectInRootView = convertFromPluginToRootView(convertFromDocumentToPlugin(convertFromPageToDocument(rectInPageSpace, *pageIndex)));
```

Comment:
A helper here would definitely make it easier to read. Probably won't need the local variable either and could return directly since the method name would hopefully make it clear

Reply:
Also fair, should just probably write out convetFromPageToRootView (and reverse)

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/45f44bb919d81d06f4f846393bce791057252d49